backend/src/chainArchive.ts

```diff
@@ -28,9 +28,7 @@ class ChainArchive {
   private readonly logger: Logger;
 
   public constructor({ path, logger }: ChainArchiveConstructorParams) {
-    this.db = levelup(rocksdb(`${path}/db`), {
-      readOnly: true,
-    });
+    this.db = levelup(rocksdb(`${path}/db`, { readOnly: true }));
```

Comment:
This is incorrect refactoring and might re-introduce bug that was fixed in #57. I see that you also added `.open()` call in `getBlocks()`, but by storing `this.db` in that case at all?